### PR TITLE
chore(SearchInput): remove type attribute, use default text type

### LINKED ISSUE #8651: SearchInput - update input type to text
**Is this a new component or an extension of an existing one?**
SearchInput

**Describe the feature**
Based off of conversation in https://github.com/patternfly/patternfly/issues/5210 and its linked PR.

tl;dr is `type="search"` for a text input can cause a browser rendered clear icon that isn't keyboard focusable to render, but not all browsers do (Firefox). Instead we should:

- use `type="text"` so that no browser clear icon renders (should just involve removing instances of `type="search"` since SearchInput defaults to text)
- ensure any examples using SearchInput have the PF clear button rendered when applicable. See [search input basic](https://www.patternfly.org/v4/components/search-input#basic) for an example of this (after typing in the input).
- Ensure examples have some sort of labeling (either text or aria) that conveys it as a search input

This would probably also include deprecating/removing the `type` prop on SearchInput to avoid dual clear icons rendering.

### PR DESCRIPTION
**Pull request by gitdallas**
Closes #8651

### CODE REVIEW

File: packages/react-core/src/demos/ComposableMenu/examples/ComposableContextSelector.tsx [hunk 1]
```diff
@@ -118,8 +118,8 @@ export const ComposableContextSelector: React.FunctionComponent = () => {
     </MenuToggle>
   );
 
-  const onSelect = (ev: React.MouseEvent<Element, MouseEvent>, itemId: string) => {
-    setSelected(itemId);
+  const onSelect = (ev: React.MouseEvent<Element, MouseEvent>, itemId: string | number) => {
+    setSelected(itemId.toString());
```
Comment (gitdallas) [inline packages/react-core/src/demos/ComposableMenu/examples/ComposableContextSelector.tsx:122]: fixed as i came across a red squiggly on the `Menu.onSelect` type mismatch (line 150)

**Review by nicolethoen — COMMENTED** [review]
In some of these cases, the `type=search` you remove is on a plain 'input' element, not the SearchInput component. I think in those cases, we probably want to leave the type=search, right? And we might consider in the future updating all of them to be SearchInputs, but that'd be out of scope for this issue.

**Comment by thatblindgeye** [comment]
> In some of these cases, the type=search you remove is on a plain 'input' element, not the SearchInput component. I think in those cases, we probably want to leave the type=search, right? And we might consider in the future updating all of them to be SearchInputs, but that'd be out of scope for this issue. 

@nicolethoen  Originally I was thinking that updating any `type="search"` to "text" for consistency/avoiding any confusion as to why SearchInput uses "text" but some other inputs/TextInputs use "search".

Your comment from the linked issue, though ("So as long as it still has a close button that is added by patternfly...") makes senss since these inputs won't have any clickable means to clear the input without `type="search"`. 

I think it'd really just depend on whether we'd get to this followup before v5 is released. If so then it probably doesn't hurt to have these `type="search"` removed from the inputs here, though we may want to make a comment to replace with SearchInput so they're easier to keep track of.

If this followup would be post v5 work, then we should keep `type="search'` on these non-SearchInput elements/components.

**Comment by nicolethoen** [comment]
I think updating those plain inputs to instead use the SearchInput would be considered a breaking change - but if it has not been done yet in core, then we shouldn't worry about doing that now. This issue was opened to make sure that there is no double 'x' close button on our SearchInputs, so I think for now, to keep this from scope creep, we should only update the SearchInput component - not plain 'input' elements.

File: packages/react-core/src/demos/PrimaryDetail.md [hunk 2]
```diff
@@ -205,7 +205,6 @@ class PrimaryDetailFullPage extends React.Component {
             <TextInput
               name="full-page-data-toolbar-input1"
               id="full-page-data-toolbar-input1"
-              type="search"
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:208]: @nicolethoen I imagine we'd want to keep these `type="search"` as well for TextInput components for now?

Comment (gitdallas) [reply packages/react-core/src/demos/PrimaryDetail.md]: added those back

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**